omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -195,8 +198,24 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					}
 					
 					if (Context.hasPrivilege(PrivilegeConstants.GET_OBS)) {
-						List<Obs> patientObs = Context.getObsService().getObservationsByPerson(p);
-						model.put("patientObs", patientObs);
+						// Get pagination parameters
+						Integer page = getPageParameter(request);
+						Integer pageSize = getPageSizeParameter(request, as);
+						
+						// Get all observations
```

Comment:
For each time that one moves through the pages, you are still fetching and loading all the patient's observations in memory. My impression was that you would load in memory only those observations for the current page.